apps/survey/contracts/Survey.sol

```diff
@@ -0,0 +1,186 @@
+        string metadata;
+
+        mapping (uint256 => uint256) votes;     // option -> voting power for option
+        mapping (address => uint256) voters;    // voter -> option voted
+    }
+
+    Survey[] surveys;
+
+    event StartSurvey(uint256 indexed surveyId);
+    event CastVote(uint256 indexed surveyId, address indexed voter, uint256 option, uint256 stake);
+    event ChangeMinParticipation(uint256 minAcceptParticipationPct);
+
+    /**
+    * @notice Initializes Survey app with `_token.symbol(): string` for governance, minimum acceptance participation of `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16` and durations of `(_surveyTime - _surveyTime % 86400) / 86400` day `_surveyTime >= 172800 ? 's' : ''`
+    * @param _token MiniMeToken address that will be used as governance token
+    * @param _minAcceptParticipationPct Percentage of total voting power that must participate in a survey for it to be taken into account (expressed as a 10^18 percentage, (eg 10^16 = 1%, 10^18 = 100%)
+    * @param _surveyTime Seconds that a survey will be open for token holders to vote
+    */
+    function initialize(
+        MiniMeToken _token,
+        uint256 _minAcceptParticipationPct,
+        uint64 _surveyTime
+    ) onlyInit external
+    {
+        initialized();
+
+        require(_minAcceptParticipationPct > 0);
+        require(_minAcceptParticipationPct <= PCT_BASE);
+
+        token = _token;
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+        surveyTime = _surveyTime;
+
+        surveys.length += 1;
+    }
+
+    /**
+    * @notice Change minimum acceptance participation to `(_minAcceptParticipationPct - _minAcceptParticipationPct % 10^14) / 10^16`%
+    * @param _minAcceptParticipationPct New acceptance participation
+    */
+    function changeMinAcceptParticipationPct(uint256 _minAcceptParticipationPct) authP(MODIFY_PARTICIPATION_ROLE, arr(_minAcceptParticipationPct)) external {
+        require(_minAcceptParticipationPct > 0);
+        require(_minAcceptParticipationPct <= PCT_BASE);
+        minAcceptParticipationPct = _minAcceptParticipationPct;
+
+        ChangeMinParticipation(_minAcceptParticipationPct);
+    }
+
+    /**
+    * @notice Create a new non-binding survey about "`_metadata`"
+    * @param _metadata Survey metadata
+    * @param _options Number of options voters can decide between
+    * @return surveyId id for newly created survey
+    */
+    function newSurvey(string _metadata, uint256 _options) auth(CREATE_SURVEYS_ROLE) external returns (uint256 surveyId) {
+        return _newSurvey(_metadata, _options);
+    }
+
+    /**
+    * @notice Vote option #`_option` in survey #`_surveyId`.
+    * @dev If voting option is 0, that's like cancelling the previously casted vote
+    * @param _surveyId Id for survey
+    * @param _option Index of supported option
+    */
+    function voteOption(uint256 _surveyId, uint256 _option) public {
```

Comment:
Should allow to specify how many tokens the voter is casting for the option. We should also have a multi-cast function so it can be done in just one tx.

See general review comment for more context.